pkg/ddl/executor.go

```diff
@@ -9627,8 +9743,21 @@ func (d *ddl) AlterCheckConstraint(ctx sessionctx.Context, ti ast.Ident, constrN
 		SQLMode:        ctx.GetSessionVars().SQLMode,
 	}
 
-	err = d.DoDDLJob(ctx, job)
-	err = d.callHookOnChanged(job, err)
+	err = e.DoDDLJob(ctx, job)
+	err = e.callHookOnChanged(job, err)
+	return errors.Trace(err)
+}
+
+// TODO remove it later.
+func (e *executor) callHookOnChanged(job *model.Job, err error) error {
```

Comment:
duplicate with what's in `ddl`, will change it later, this pr is very large now.